### ISSUE TITLE
allow specifying cacerts

### ISSUE THREAD

**Issue by kapilt** [httplib2/httplib2#114]
currently httplib2 embeds its cert authority bundle, in cases where an organization has an additional certs authorities, it would be good to allow a user/app to specify a different ca bundle.

**Comment by temoto** [comment]
Yes you can specify another cert bundle. The issue is that it doesn't work
or needs more highlight in documentation?

**Comment by kapilt** [comment]
its little unwieldy for users of application using httplib2, as they would need to modify the app usage of httplib2 to override the default ca bundle. As an example for `requests`, users can simply set an environment variable without modification of a third party application.

Within httplib2 it appears that some effort to address that was applied to python2, as it has a conditional import of a user supplied/magic module [0], though that mechanism I find a bit cumbersome to document for end users of an app. That behavior is not present in the python3 implementation.

I'd suggest just adding a HTTPLIB2_CA_BUNDLE environment variable to allow for this default ca bundle override.

[0] - https://github.com/httplib2/httplib2/blob/master/python2/httplib2/__init__.py#L276

**Comment by temoto** [comment]
I agree, user supplied module is cumbersome. Don't use it.

Does this work for you?

```python
http = httplib2.Http(ca_certs=os.environ.get('HTTPLIB2_CA_BUNDLE'))
http.request('https://custom-tls-server')
```

**Comment by kapilt** [comment]
That does work for users directly coding against httplib2, the issue with direct configuration is that for users of third-party applications built on top of httplib2, that may not be a viable option without modifying the third-party application source. Providing  the env var configuration directly within the library allows those users to do so without source modifications to third party apps. ie. this is why `requests` provides such an env var mechanism to configure ca bundle.

**Comment by temoto** [comment]
@kapilt so we have a working solution, that's a great first step.

Could you please point to piece of code in requests which handles CA via environ, because I tried and could not find it.

**Comment by kapilt** [comment]
It’s in
https://github.com/requests/requests/blob/master/requests/sessions.py
within the merge_environment_settings function

**Comment by kapilt** [comment]
addressed and merged in #117
